reportgenerator.py

```diff
@@ -0,0 +1,217 @@
+
+        The returned object is a dictionary of dictionaries that associates a service with a dictionary associating
+        dates and costs.
+
+        ex:
+            {
+                'username1' : {
+                    'EC2':
+                        {
+                            '2018-12-30': 126.45,
+                            '2018-12-31': 60.45,
+                            'Total': 186.90
+                        }, ...,
+                    'Total': 326.95
+                    },
+
+                'username2' : {
+                    'EC2':
+                        {
+                            '2018-12-30': 365.63,
+                            '2018-12-31': 100.00,
+                            'Total': 465.63
+                        }, ...,
+                    'Total': 763.23
+                },...,
+
+                'Total': 12345.33
+
+            }
+
+        :param dict response: The response from the AWS Cost Explorer API.
+        :returns defaultdict(defaultdict(dict)) processed: Data from the response organized by service:date:cost.
+        """
+        processed = defaultdict(lambda: defaultdict(dict))
+
+        daily_data = response['ResultsByTime']
+
+        # Create dict associating dates and costs, ordered by service.
+        for day_dict in daily_data:
+            date = day_dict['TimePeriod']['Start']
+
+            services_used = day_dict['Groups']
+            for s in services_used:
+                owner = s['Keys'][0].split('$')[1]
+                service = s['Keys'][1]
+                cost = float(s['Metrics']['BlendedCost']['Amount'])
+
+                processed[owner][service][date] = cost
+
+        # Calculate totals for each owner, service and overall.
+        everyone_total = 0.0
+        for owner in processed:
+            owner_total = 0.0
+
+            for service in processed[owner]:
+                service_total = 0.0
+
+                for cost in processed[owner][service].values():
+                    service_total += cost
+                    owner_total += cost
+                    everyone_total += cost
+
+                processed[owner][service]['Total'] = service_total
+            processed[owner]['Total'] = owner_total
+        processed['Total'] = everyone_total
+
+        return processed
+
+    def send_everyone_report(self):
+        """
+        Print a report summarizing all of the data.
+
+        :return dict response: The response.
+        """
+        pass
+
+    def send_individual_report(self, user):
+        """
+        Print a report for a given user.
+
+        :param str user: The email address of the user who the report is about.
+        """
+        response_by_account = dict()
+        for acct_num in self.account_nums:
+            response = self.api_call([user], [acct_num])
+            processed = self.process_api_response(response)
+            response_by_account[acct_num] = processed
+
+        print('Report for {}'.format(user))
```

Comment:
Make this section into a separate function that returns the report as a string to send?